### DISCUSSION TITLE
[BETA] First beta version of 0.3.1 released!

### DISCUSSION THREAD

**Discussion by zhenrong-wang** [zhenrong-wang/hpc-now#23] [Announcements]
**Information security is the top issue.**
**IMPORTANT: This upgrade is not compatible with previous versions, that is, 0.2.x**

----------------------------------

**Main Changes: Strongly enhanced the information security by introducing AES-128 encryption**

The **HPC-NOW** services handles sensitive information, typically are cloud access key (known as AK) and access secret key(known as SK), or key files (for Google Cloud Platform). Meanwhile, when managing clusters in the cloud, more sensitive information such as passwords are generated. All the information mentioned **MUST** be handled and protected carefully.

Since Day One of this project, we have been considering information security. That's why we have an independent crypto module named `now-crypto` for the running of **HPC-NOW** services ever since the first release of this project. However, the previous crypto module has been a weak point, because it is simple, and a not standard one. 

Therefore, after days of work, we introduced a genuine, data block level, industrial standard, wide-recognized crypto module - **AES-128 (ECB mode) with PKCS padding**. With the new crypto module, the security level of **HPC-NOW** really meets the requirement of production. In our development plan, this AES-based module will be a long-term part of **HPC-NOW**.

We strongly recommend you start your HPC-NOW journey from this version - **0.3.x**

And, **VERY IMPORTANT: DO NOT directly update from 0.2.x to 0.3.x**. If you'd like to update, please submit issues to us. We are also developing a utility to make the update safe and smooth.

-----------------------------------------------------
*We also added a CONTRIBUTING.md to this project, looking forward to your feedback and contributions!*

*HPC-NOW: Start your HPC journey in the cloud now.*

<hr /><em>This discussion was created from the release <a href='https://github.com/zhenrong-wang/hpc-now/releases/tag/0.3.1.beta1'>[BETA] First beta version of 0.3.1 released!</a>.</em>

**Comment by tracyzhu019** [comment]
Will the dev0.2.1 be upgraded to 0.3.1?

**Reply by zhenrong-wang** [reply]
Please do not update directly, we will introduce an update package special for 0.2.1 users to update safely. 

Please stay tuned.
